librosa/core/spectrum.py

```diff
@@ -519,6 +520,450 @@ def ifgram(y, sr=22050, n_fft=2048, hop_length=None, win_length=None,
+
+    The reassignment vector is calculated using equation 5.20 in Flandrin,
+    Auger, & Chassande-Mottin 2002:
+
+    .. math:: \omega - \Im(\frac{S_{dh}}{S_h})
+
+    where `S_h` is the complex STFT calculated using the original window, and
+    `S_dh` is the complex STFT calculated using the derivative of the original
+    window.
+
+    See `reassigned_spectrogram` for references.
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,)], real-valued
+        audio time series
+
+    sr : number > 0 [scalar]
+        sampling rate of `y`
+
+    n_fft : int > 0 [scalar]
+        FFT window size
+
+    hop_length : int > 0 [scalar]
+        hop length, number samples between subsequent frames.
+        If not supplied, defaults to `win_length / 4`.
+
+    win_length : int > 0, <= n_fft
+        Window length. Defaults to `n_fft`.
+        See `stft` for details.
+
+    window : string, tuple, number, function, or np.ndarray [shape=(n_fft,)]
+        - a window specification (string, tuple, number);
+          see `scipy.signal.get_window`
+        - a window function, such as `scipy.signal.hanning`
+        - a user-specified window vector of length `n_fft`
+
+        See `stft` for details.
+
+        .. see also:: `filters.get_window`
+
+    dtype : numeric type
+        Complex numeric type for `S`. Default is 64-bit complex.
+
+    Returns
+    -------
+    freqs : np.ndarray [shape=(1 + n_fft/2, t), dtype=real]
+        Instantaneous frequencies:
+        `freqs[f, t]` is the frequency for bin `f`, frame `t`
+
+    S : np.ndarray [shape=(1 + n_fft/2, t), dtype=complex]
+        Short-time Fourier transform
+
+    Warns
+    --------
+    RuntimeWarning
+        Frequencies with zero support will produce a divide-by-zero warning and
+        will be returned as `np.nan`.
+
+    See Also
+    --------
+    stft : Short-time Fourier Transform
+    reassign_times : Estimate time reassignments
+    reassigned_spectrogram : Time-frequency reassigned spectrogram
+
+    Examples
+    --------
+    >>> y, sr = librosa.load(librosa.util.example_audio_file())
+    >>> frequencies, S = librosa.reassign_frequencies(y, sr=sr)
+    >>> frequencies
+    array([[  0.000e+00, 0.000e+00, ..., 0.000e+00, 0.000e+00],
+           [  9.863e+00, 9.653e+00, ..., 1.046e+01, 8.072e+01],
+           ...,
+           [  1.101e+04, 1.101e+04, ..., 1.102e+04, 1.102e+04],
+           [  1.102e+04, 1.102e+04, ..., 1.102e+04, 1.102e+04]])
+
+    """
+
+    # retrieve window samples if needed so that the window derivative can be
+    # calculated
+    if win_length is None:
+        win_length = n_fft
+
+    window = get_window(window, win_length, fftbins=True)
+    window = util.pad_center(window, n_fft)
+
+    # center is always False for consistency with `reassign_times`
+    S_h = stft(
+        y=y,
+        n_fft=n_fft,
+        hop_length=hop_length,
+        window=window,
+        center=False,
```

Comment:
I see where you're coming from here, but I think it's better to have consistency with the entire API (center=True by default) and document the boundary effects.  In general, this should be a parameter that the user can control.

Reply:
Done in https://github.com/librosa/librosa/pull/926/commits/9450b92e021de9f34cb5044957d9501c61098f20

Reply:
I put notes about this in the function docstring summaries. Is that the right place for it?